### PR TITLE
Map/update delete to poi

### PR DESCRIPTION
**Pull request by yevhenverkhalantsev**

### CODE REVIEW

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/maps_settings/adapter/PointOfInterestRecyclerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.dashcarr.R
+import com.example.dashcarr.domain.entity.PointOfInterestEntity
+
+class PointOfInterestRecyclerAdapter (private val onPointListener: OnPointClickListener)
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/maps_settings/adapter/PointOfInterestRecyclerAdapter.kt:14]: That seems like a really important concept for working with the database. I don't know it and I assume the others does't now it as well. Can you give us tomorrow a short introduction for the best practices if you want to read/write to the database?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -123,4 +123,14 @@
...
 
+    <!--MapSettingsFragment-->
+
+    <string name="points_of_interest">Points Of Interest</string>
+    <string name="btnRename">Rename</string>
+    <string name="btnDelete">delete</string>
+    <string name="update_poi">update</string>
+    <string name="delete_poi">delete</string>
+    <string name="point_deleted_successfully">Point deleted successfully.</string>
+    <string name="point_renamed_successfully">Point renamed successfully.</string>
```
Comment (ysz0507) [inline app/src/main/res/values/strings.xml:134]: "Point deleted successfully." is not a sentence, so it also does not need a dot.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -123,4 +123,14 @@
...
     <string name="search_for_location">Waiting for GPS</string>
 
+    <!--MapSettingsFragment-->
+
+    <string name="points_of_interest">Points Of Interest</string>
+    <string name="btnRename">Rename</string>
+    <string name="btnDelete">delete</string>
+    <string name="update_poi">update</string>
+    <string name="delete_poi">delete</string>
+    <string name="point_deleted_successfully">Point deleted successfully.</string>
```
Comment (ysz0507) [inline app/src/main/res/values/strings.xml:133]: same here

**Review by ysz0507 — APPROVED** [review]
Nice work! Most important is the bug, that the camera is not following the car anymore after centering. 
You did not create a new branch from the master in the correct way after you finished working on the centering button. So now both features are in one branch. For now not a big deal but pay attention for the next time :)

File: app/src/main/java/com/example/dashcarr/data/datasource/points/IPointsOfInterestLocalDataSource.kt
[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/data/datasource/points/IPointsOfInterestLocalDataSource.kt:1]: it's not possible as I know, cause scroll view filling up with exist content and it means that marker dialog will be always in center of height current view. and scroll view can have inside only one child, so I can't make another constraint either.

[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/data/datasource/points/IPointsOfInterestLocalDataSource.kt:1]: found solution! over scroll having new constraint as head, and setting my dialogs below scroll view and they get position forward scroll and constrained to parent.

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/maps_settings/MapsSettingsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    _onRenamePointSuccess.emit(Unit)
+                }
+                result.onFailure {
+                    _onRenamePointFailure.emit(it)
+                }
+            }
+        }
+    }
+
+    // Function to delete a point of interest
```
[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/presentation/tabs/settings/maps_settings/MapsSettingsViewModel.kt:52]: added kdoc to viemodel

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (yevhenverkhalantsev) [reply app/src/main/res/values/strings.xml:134]: removed dots

File: app/src/main/java/com/example/dashcarr/data/datasource/points/IPointsOfInterestLocalDataSource.kt
[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/data/datasource/points/IPointsOfInterestLocalDataSource.kt:1]: I will fix it during next tasks. Now it's too many changes in map fragment where is not only my code.

[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/data/datasource/points/IPointsOfInterestLocalDataSource.kt:1]: now its Map Settings

File: app/src/main/java/com/example/dashcarr/presentation/tabs/map/MapFragment.kt [hunk 5]
```diff
@@ -66,6 +67,8 @@ class MapFragment : BaseFragment<FragmentMapBinding>(
 
     // Location manager for handling location updates
     private val locationManager by lazy { requireContext().getSystemService(Context.LOCATION_SERVICE) as LocationManager }
+    //private val myLocationOverlay by lazy { MyLocationNewOverlay(binding.mapView) }
```
[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/presentation/tabs/map/MapFragment.kt]: done

File: app/src/main/java/com/example/dashcarr/presentation/tabs/map/MapFragment.kt [hunk 6]
```diff
@@ -490,10 +497,11 @@ class MapFragment : BaseFragment<FragmentMapBinding>(
...
         locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 2000, 100f, this)
         locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 2000, 100f, this)
-        val overlay = MyLocationNewOverlay(binding.mapView)
-        overlay.enableMyLocation()
-        overlay.enableFollowLocation()
-        binding.mapView.overlays.add(overlay)
+        myLocationOverlay = MyLocationNewOverlay(binding.mapView)
+        myLocationOverlay.enableMyLocation()
+        myLocationOverlay.enableFollowLocation()
+//        TODO: Bug on back click to map fragment.
```
[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/presentation/tabs/map/MapFragment.kt]: done

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/maps_settings/MapsSettingsViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    private val _onDeletePointSuccess = MutableSharedFlow<Unit>()
+    val onDeletePointSuccess = _onDeletePointSuccess.asSharedFlow()
+    private val _onDeletePointFailure = MutableSharedFlow<Throwable>()
+    val onDeletePointFailure = _onDeletePointFailure.asSharedFlow()
+    private val _onRenamePointSuccess = MutableSharedFlow<Unit>()
+    val onRenamePointSuccess = _onRenamePointSuccess.asSharedFlow()
+    private val _onRenamePointFailure = MutableSharedFlow<Throwable>()
+    val onRenamePointFailure = _onRenamePointFailure.asSharedFlow()
+
+    // Function to rename a point of interest
```
[the remark this answers is not in the document]
Comment (yevhenverkhalantsev) [reply app/src/main/java/com/example/dashcarr/presentation/tabs/settings/maps_settings/MapsSettingsViewModel.kt]: done

**Pull request merged**